pandas/core/frame.py

```diff
@@ -9803,12 +9803,16 @@ def apply(
         )
         return op.apply().__finalize__(self, method="apply")
 
-    def applymap(
+    def map(
         self, func: PythonFuncType, na_action: str | None = None, **kwargs
```

Comment:
This is really an aside, but I wanted to mention as we do have the opportunity here. In the cases where we're passing through kwargs to a user-passed function, I have increasingly become a fan of having `kwargs` be a normal argument as opposed to `**kwargs`. While these points are relatively minor, the benefit is:

 - No possibility of name conflicts between the pandas method and the user passed function (e.g. `func` can't have an argument `na_action` here)
 - Typos (e.g. `df.map(n_action='ignore')`) generate better error messages

One possible downside is users have to do `df.map(foo, kwargs={'a': 1})` instead of `df.map(foo, a=1)` but I personally don't think that as much of a downside.

Reply:
I agree with all your points personally and I try not to use `*args` or `**kwargs` at all. Also, if you look at e.g. numpy, it uses  `*args` or `**kwargs` alot, and I think it makes numpy harder to use than it needs to.

OTOH, there is social expectations and people are used to  `*args` or `**kwargs`, so i don't know, could be a too big change now that we already have selected the  `*args` or `**kwargs` route.

There is though also [ParamSpec](https://peps.python.org/pep-0612/), which we could implement to partly alleviate this problem.